### DISCUSSION TITLE
TextInput mask decimalSeparator V3

### DISCUSSION THREAD

**Discussion by AFSDARIF** [filamentphp/filament#8066] [Help]
How to fix this problem after upgrade to V3.
![image](https://github.com/filamentphp/filament/assets/10229660/e16e49cf-0571-43fc-a09d-c10cd4ab7ccf)

```
->numeric()
->mask(fn (Forms\Components\TextInput\Mask $mask) => $mask
    ->numeric()
    ->decimalSeparator('.'),
)
```

**Comment by micraux** [comment]
Have you try setting the HTML input mode to `decimal`?
https://filamentphp.com/docs/3.x/forms/fields/text-input#setting-the-html-input-mode
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode

**Reply by AFSDARIF** [reply]
but, how replace decimal coma , with dot . ?
i want 12.54
not 12,54

**Reply by danharrin** [reply]
It depends on the browsers locale

**Comment by AFSDARIF** [comment]
i think i have found the solution with javascript

```
->mask(RawJs::make(<<<'JS'
    $input.replace(/[^0-9.]/g, '').replace(/(\..*?)\..*/g, '$1');
JS))
```
